---
version: "3.8"

services:
  traefik:
    image: traefik:v2.6
    ports:
      - "${PROJECT_PORT_EXTERNAL}:80"
    networks:
      - traefik-dev-proxy
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock:ro"
      - "./traefik.dev.yml:/etc/traefik/traefik.yml"
    labels:
      - "traefik.enable=true"

  django:
    command: >
      bash -c "./1--secret-key-create-if-not-exists &&
               ./2--server-config-choose dev &&
               python3 manage.py migrate &&
               python3 manage.py runserver 0.0.0.0:${PROJECT_PORT_INTERNAL}"
    expose:
      - "${PROJECT_PORT_INTERNAL}"
    networks:
      - traefik-dev-proxy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.${PROJECT_NAME}.rule=Host(`${DEV_DOMAIN_NAME}`, `www.${DEV_DOMAIN_NAME}`)"
      - "traefik.http.routers.${PROJECT_NAME}.entrypoints=web"
    depends_on:
      - traefik

networks:
  traefik-dev-proxy:
    driver: bridge
